packages/eas-cli/src/update/queries.ts

```diff
@@ -2,84 +2,180 @@ import chalk from 'chalk';
+      offset: paginatedQueryOptions.offset,
+      appId: projectId,
+      branchName,
+    });
+    renderUpdateGroupsAsTable(updateGroups, paginatedQueryOptions);
+  } else {
+    await paginatedQueryWithConfirmPromptAsync({
+      limit: paginatedQueryOptions.limit ?? UPDATE_GROUPS_LIMIT,
+      offset: paginatedQueryOptions.offset,
+      queryToPerform: (limit, offset) =>
+        queryUpdateGroupsOnBranchAsync({ limit, offset, appId: projectId, branchName }),
+      promptOptions: {
+        title: 'Load more update groups?',
+        renderListItems: updates => renderUpdateGroupsAsTable(updates, paginatedQueryOptions),
+      },
+    });
+  }
+}
+
+export async function selectUpdateGroupOnBranchAsync({
+  projectId,
+  branchName,
+  paginatedQueryOptions: painatedQueryOptions,
+}: {
+  projectId: string;
+  branchName: string;
+  paginatedQueryOptions: PaginatedQueryOptions;
+}): Promise<BranchUpdateGroupObject> {
+  if (painatedQueryOptions.nonInteractive) {
+    throw new Error('Unable to select an update in non-interactive mode.');
+  } else {
+    const updateGroup = await paginatedQueryWithSelectPromptAsync({
+      limit: painatedQueryOptions.limit ?? UPDATE_GROUPS_LIMIT,
+      offset: painatedQueryOptions.offset,
+      queryToPerform: (limit, offset) =>
+        queryUpdateGroupsOnBranchAsync({ appId: projectId, branchName, limit, offset }),
+      promptOptions: {
+        title: 'Load more update groups?',
+        createDisplayTextForSelectionPromptListItem: updateGroup =>
+          formatUpdateTitle(updateGroup[0]),
+        getIdentifierForQueryItem: updateGroup => updateGroup[0].group,
+      },
+    });
+
+    if (!updateGroup || updateGroup.length === 0) {
+      throw new Error(`Could not find any branches for project "${projectId}"`);
+    }
+
+    return updateGroup;
+  }
+}
+
+async function queryUpdateGroupsOnBranchAsync(
+  args: ViewUpdateGroupsOnBranchForAppQueryVariables
+): Promise<BranchUpdateGroupObject[]> {
+  const { app } = await UpdateQuery.viewUpdateGroupsOnBranchAsync(args);
+
+  const { updateBranchByName } = app.byId;
+  if (!updateBranchByName) {
+    throw new Error(`Could not find branch "${args.branchName}"`);
+  }
+
+  return updateBranchByName.updateGroups;
+}
+
+async function queryUpdateGroupsOnAppAsync({
+  limit,
+  offset,
+  appId,
+}: {
+  limit: number;
+  offset: number;
+  appId: string;
+}): Promise<AppUpdateGroupObject[]> {
+  const { app } = await UpdateQuery.viewUpdateGroupsAsync({
+    appId,
     limit,
     offset,
   });
 
-  const updateBranch = app?.byId.updateBranchByName;
-  if (!updateBranch) {
-    throw new Error(`Could not find branch "${branchName}"`);
+  if (!app) {
+    throw new Error(`Could not find app with id "${appId}"`);
   }
 
-  return updateBranch.updates;
+  return app.byId.updateGroups;
 }
 
-function renderUpdateGroups(
-  updates: BranchUpdateObject[],
-  { json }: PaginatedQueryOptions,
-  branchName: string
+function renderUpdateGroupsAsTable(
```

Comment:
add a docblock (or runtime assertion might be even better) that checks that all updateGroups passed into this are for the same branch

Reply:
cool

Reply:
branch without updates:
![image](https://user-images.githubusercontent.com/15132641/188227900-3692b1a3-7eff-404b-a84c-ccc375c62743.png)

branch with updates:
![image](https://user-images.githubusercontent.com/15132641/188227979-13eccfc8-dbe5-4be0-a30d-94111859e952.png)